XenAdminTests/XenModelTests/AreEqual2Tests.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System.Collections.Generic;
+using NUnit.Framework;
+using XenAPI;
+
+namespace XenAdminTests.XenModelTests
+{
+    [TestFixture, Category(TestCategories.Unit)]
+    public class AreEqual2Tests
+    {
+        [Test]
+        public void A_null_is_considered_equal_to_itself()
+        {
+            List<string> given = null;
+            Assert.That(Helper.AreEqual2(given, given), Is.True);
+        }
+
+        [Test]
+        public void Separate_nulls_are_considered_equal_to_each_other()
+        {
+            Assert.That(Helper.AreEqual2((List<string>)null, (List<string>)null), Is.True);

```

Comment:
Let's add one more assertion for null Dictionaries since they are a class used heavily in the bindings.
Also, an important test case missing is comparison of null List with empty List and null Dictionary with empty Dictionary (which is what differentiates the method AreEqual2 from AreEqual).

Reply:
Agreed, its better to be thorough.